packages/synapse-sdk/src/storage/manager.ts

```diff
@@ -146,26 +159,55 @@ export class StorageManager {
       }
     }
 
-    // Get the context to use
+    // Get the context(s) to use
     const contexts = options?.contexts ?? [options?.context ?? (await this.createContext(options))]
-    // Convert once
-    const dataBytes = data instanceof ArrayBuffer ? new Uint8Array(data) : data
-    // Calculate pieceCid once
-    const pieceCid = Piece.calculate(dataBytes)
-
-    // Upload using the contexts with piece metadata
-    return await Promise.allSettled(
-      contexts.map((context) =>
-        context.upload(
-          dataBytes,
-          {
+
+    // Multi-context upload handling
+    if (contexts.length > 1) {
+      // Multi-context uploads require Uint8Array to calculate pieceCid once
+      if (!(data instanceof Uint8Array)) {
+        throw createError(
+          'StorageManager',
+          'upload',
+          'Multi-context uploads currently only support Uint8Array. ' +
+            'For streaming uploads to multiple providers, convert your stream to Uint8Array first.'
+        )
```

Comment:
i'm assuming this is a temporary solution for the forking-stream + streaming-pieceCid-calculation problem?

The reason i'm asking is i'm curious what folks' solutions' are for getting around this once merged? Things i'm thinking:

1. convert to uint8array as error states..
2. perform multiple explicit upload calls with your stream (but they still need to calculate pieceCid in this case, right?)